include/scrimmage/simcontrol/SimControl.h

```diff
@@ -64,64 +64,215 @@ enum class EndConditionFlags {TIME = 1, ONE_TEAM = 2, NONE = 3, ALL_DEAD = 4};
+     *  \endcode
+     */
+    bool run_single_step(const int& loop_number);
+
+    /**
+     * @brief Finalizes the simulation, closes logs, closes plugins.
+     *
+     * The shutdown() function should be called after the simulation is
+     * complete.
+     */
+    bool shutdown();
+
+    /**
+     * @brief Force a threaded simulation to exit.
+     *
+     * When running the simulation in a separate thread, this function sends a
+     * signal to the main simulation loop to exit.
+     */
     void force_exit();
-    bool external_exit();
+
+    /**
+     * @brief Waits for the threaded simulation to exit.
+     */
     void join();
 
-    bool finished();
-    void set_finished(bool finished);
+
+    /*
+    * @brief Provides access to the SimControl's entity plugin.
+    *
+    * After SimControl has been initialized with the init() function, this
+    * function provides access to the entity plugin owned by SimControl. This
+    * entity plugin can be used to access the publish/subscribe bus used during
+    * the simulation. For example, when used during a test, the test can
+    * publish and subscribe to messages used by plugins under test.
+    */
+    EntityPluginPtr plugin();
 
     void get_contacts(std::unordered_map<int, Contact> &contacts);
-    void set_contacts(ContactMapPtr &contacts);
 
     void get_contact_visuals(std::map<int, ContactVisualPtr> &contact_visuals);
-    void set_contact_visuals(std::map<int, ContactVisualPtr> &contact_visuals);
 
-    bool generate_entities(double t);
+    /**
+     * @brief Generate entities based on the current time.
+     *
+     * @param [in] t Current simulation time.
+     *
+     * Determines the entities that should be generated at the given time and
+     * generates the entities.
+     */
+    bool generate_entities(const double& t);
+
+    /**
+     * @brief Generate an entity given the entity description ID.
+     */
     bool generate_entity(const int &ent_desc_id);
+
+    /**
+     * @brief Generate an entity given the entity description ID and
+     * parameters.
+     */
     bool generate_entity(const int &ent_desc_id,
                          std::map<std::string, std::string> &params);
 
-    void set_mission_parse(MissionParsePtr mp);
+    /**
+     * @brief Get the pointer to the MissionParser instance.
+     *
+     */
     MissionParsePtr mp();
-    void set_log(std::shared_ptr<Log> &log);
 
-    bool enable_gui();
+    /**
+     * @brief Set the simulation time
+     *
+     * @param [in] t The simulation time
+     */
+    void set_time(const double &t);
 
-    void set_time(double t);
+    /**
+     * @brief Get the current simulation time.
+     */
     double t();
 
-    bool output_summary();
-    bool output_runtime();
-    void setup_timer(double rate, double time_warp);
-    void start_overall_timer();
-    void start_loop_timer();
-    void loop_wait();
-    void inc_warp();
-    void dec_warp();
-    void pause(bool pause);
+    /**
```

Comment:
it seems like all of this comment and the others could just be a single line. doxygen formatting allows this with `///`

Reply:
Ok. Single line doxygen comments use /// now.